src/modules/accounting/third_phase/extra_data_v2.py

```diff
@@ -0,0 +1,146 @@
+
+        if items_count:
+            extra_data_format = FormatList.EXTRA_DATA_FORMAT_LIST_NON_EMPTY
+        else:
+            extra_data_format = FormatList.EXTRA_DATA_FORMAT_LIST_EMPTY
+
+        return ExtraData(
+            items_count=items_count,
+            extra_data_list=hashed_txs,
+            data_hash=first_hash,
+            format=extra_data_format,
+        )
+
+    @classmethod
+    def build_validators_payloads(
+        cls,
+        validators: dict[NodeOperatorGlobalIndex, int],
+        max_no_in_payload_count: int,
+    ) -> list[ItemPayload]:
+        operator_validators = sorted(validators.items(), key=lambda x: x[0])
+
+        payloads = []
+
+        for module_id, operators_by_module in groupby(operator_validators, key=lambda x: x[0][0]):
+            for nos_in_batch in batch(list(operators_by_module), max_no_in_payload_count):
+                operator_ids = []
+                vals_count = []
+
+                for ((_, no_id), validators_count) in nos_in_batch:
+                    operator_ids.append(no_id)
+                    vals_count.append(validators_count)
+
+                payloads.append(
+                    ItemPayload(
+                        module_id=module_id,
+                        node_operator_ids=operator_ids,
+                        vals_counts=vals_count,
+                    )
+                )
+
+        return payloads
+
+    @classmethod
+    def build_extra_transactions_data(
+        cls,
+        stuck_payloads: list[ItemPayload],
+        exited_payloads: list[ItemPayload],
+        max_items_count: int,
+    ) -> tuple[int, list[bytes]]:
+        all_payloads = [
+            *[(ItemType.EXTRA_DATA_TYPE_STUCK_VALIDATORS, payload) for payload in stuck_payloads],
+            *[(ItemType.EXTRA_DATA_TYPE_EXITED_VALIDATORS, payload) for payload in exited_payloads],
+        ]
+
+        index = 0
+        result = []
+
+        for payload_batch in batch(all_payloads, max_items_count):
+            tx = b''
+            for item_type, payload in payload_batch:
+                tx += index.to_bytes(ExtraDataLengths.ITEM_INDEX)
+                tx += item_type.value.to_bytes(ExtraDataLengths.ITEM_TYPE)
+                tx += payload.module_id.to_bytes(ExtraDataLengths.MODULE_ID)
+                tx += len(payload.node_operator_ids).to_bytes(ExtraDataLengths.NODE_OPS_COUNT)
+                tx += b''.join(
+                    no_id.to_bytes(ExtraDataLengths.NODE_OPERATOR_IDS)
+                    for no_id in payload.node_operator_ids
+                )
+                tx += b''.join(
+                    count.to_bytes(ExtraDataLengths.STUCK_OR_EXITED_VALS_COUNT)
+                    for count in payload.vals_counts
+                )
+
+                index += 1
+
+            result.append(tx)
+
+        return index, result
+
+    @staticmethod
+    def add_hashes_to_transactions(txs_data: list[bytes]) -> tuple[bytes, list[bytes]]:
+        txs_data.reverse()
```

Comment:
why we need reverse here?

Reply:
We construct a linked list of items bound via `nextHash` field. So we start hashing from the very end to get the sequence.